lib/puppetserver/ca/action/generate.rb

```diff
@@ -77,31 +77,32 @@ def run(input)
           end
 
           # Load, resolve, and validate puppet config settings
-          puppet = Config::Puppet.parse(config_path)
+          settings_overrides = {}
+          settings_overrides[:certname] = input['certname'] unless input['certname'].empty?
+          settings_overrides[:ca_name] = input['ca_name'] unless input['ca_name'].empty?
+          settings_overrides[:subject_alt_names] = input['subject_alt_names'] unless input['subject_alt_names'].empty?
```

Comment:
Now that we have this, won't we need to change
```
settings[:subject_alt_names] = overrides.fetch(:dns_alt_names, "puppet,$certname")
```
in the puppetconfig file? Or the subject alt names will just get overwritten by dns-alt-names?

Reply:
Good catch! I think this value should be set to the key `:dns_alt_names` instead of `:subject_alt_names` so that its merged correctly within Config::Puppet.

Reply:
That sounds right, will fix.